### DISCUSSION TITLE
Recipe for adding bash to Kaniko

### DISCUSSION THREAD

**Discussion by BeyondEvil** [GoogleContainerTools/kaniko#3102] [Show and tell]
```Dockerfile
FROM curlimages/curl:8.7.1 as base

RUN curl -sSLo "bash" "https://github.com/robxu9/bash-static/releases/download/5.2.015-1.2.3-2/bash-linux-x86_64" && \
    chmod +x bash

FROM gcr.io/kaniko-project/executor:v1.21.1-debug as final

COPY --from=base /bash /busybox/
RUN ln -s /busybox/bash /bin/bash
```

Obviously, you can play around with ARGs to make the versions dynamic.

**Comment by abitrolly** [comment]
But debug image already comes with `busybox` shell - https://github.com/GoogleContainerTools/kaniko?tab=readme-ov-file#debug-image

**Reply by BeyondEvil** [reply]
Correct.

The recipe is for use cases where the busybox shell doesn't suffice.

**Reply by abitrolly** [reply]
@BeyondEvil but the recipe also packs `busybox` shell, so it makes no difference.

**Reply by abitrolly** [reply]
Ok. My bad. It is static `bash` binary that overwrites `busybox`. Still many questions. )

**Reply by BeyondEvil** [reply]
It doesn't overwrite the busybox shell. The busybox shell `/busybox/sh` is symlinked to `/bin/sh`.

The static bash binary `/busybox/bash` is symlinked to `/bin/bash`.

Ask away. I'm happy to answer to the best of my ability. :)

**Reply by abitrolly** [reply]
Why not copy without symlinks?
```Dockerfile
COPY --from=base /bash /bin/
```

**Reply by BeyondEvil** [reply]
> Why not copy without symlinks?
> 
> ```dockerfile
> COPY --from=base /bash /bin/
> ```

I can't remember, maybe that works. 🤷‍♂️ 😅
